### ISSUE TITLE
feeding ivar pre-stitched PE reads

### ISSUE THREAD

**Issue by tseemann** [andersen-lab/ivar#29]
@mcroxen and i were wondering
if you pre-stitch your PE reads (eg. FLASH, bbmerge.sh, fastp --merge)
And then align these longer SE reads and feed to ivar
Will it still work in terms of primer trimming etc?

**Comment by gkarthik** [comment]
`ivar trim` trims primers from 5' end. For stitched reads, this information would be lost. I could add another check for the 3' end of the read though. So if the 3' end of the read overlaps with a primer trim that bit off too. Autodetect stitched reads using length?

**Comment by tseemann** [comment]
Maybe if you check the BAM flags, you will be able to differentiate between paired and single-end reads?  single-end will NOT be "paired in sequencing" in anyway. If SE, then trim 5' and 3'. If PE then trim only 5' ?

```
xxxx = Illumina adaptor
PPPP = primer
ssss = sequence
SSSS = overlapping sequence

Sequenced:

R1    5' xxxxPPPPPPsssssSSS-> 3'
R2'   3'  <-SSSssssPPPPPxxxxx  5'

Adaptors removed:

R1    5' PPPPPPsssssSSS-> 3'
R2'   3'  <-SSSssssPPPPP  5'


Stitched:

OLAP  5' PPPPPPsssssSSSssssPPPPP 5'

```

**Comment by gkarthik** [comment]
I think you’re right. I can check for read not paired and then trim from 3’ end as well.

**Comment by tseemann** [comment]
That would be very nice!

**Comment by gkarthik** [comment]
Fixed by 72b35e698b737e787be1d4fa93eeb6de972aeeb0
